src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -3862,6 +3865,11 @@ public bool DisconnectAsync(SocketAsyncEventArgs e)
             // Throw if socket disposed
             ThrowIfDisposed();
 
+            if (e == null)
+            {
+                throw new ArgumentNullException(nameof(e));
+            }
```

Comment:
You may have already done this and I just haven't come to it yet, but can you make sure there's a test for this?

Reply:
No, I forgot. Thanks for the reminder.